scripts/update-versions.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+
+if [ "$#" -ne 1 ]; then
+	echo "usage: $(basename $0) <NEW_VERSION>" > /dev/stderr
+	exit 1
+fi
+version=$1
+
+toml_cli=${TOML_CLI:-${CARGO_HOME:-~/.cargo}/bin/toml}
+$toml_cli -h >/dev/null || {
+	echo "'toml' utility not found. please run \`cargo install toml-cli\`." > /dev/stderr
+	exit 1
+}
+
+# collect all .toml files
+tomls=$(find -name Cargo.toml)
+
+# collect all .toml files for non-test/example crates
+own_crate_tomls=$(find -name Cargo.toml | grep -E -v 'tests/|examples/')
+
+# generate list of crate names (whose versions as deps we should update)
+crates=$(for toml in $own_crate_tomls; do
+	$toml_cli get $toml package.name | tr -d \"
+done)
+
+update_key() {
+	toml=$1
+	key=$2
+	value=$3
+
+	old_value=$($toml_cli get $toml $key)
+	# only set value if already present
+	if [ ! -z "$old_value" ]; then
+		$toml_cli set $toml $key $value | sponge $toml
```

Comment:
What's `sponge`?

Reply:
https://manpages.debian.org/testing/moreutils/sponge.1.en.html

It avoids the pipeline truncating the .toml file to use as an output before it can be read from to use as an input.